lumicks/pylake/force_calibration/power_spectrum.py

```diff
@@ -72,37 +146,109 @@ def squared_fft(d):
         ]
 
         squared_fft = np.mean(squared_fft_chunks, axis=0)
-
-        self.frequency = np.fft.rfftfreq(num_points_per_window, 1.0 / sample_rate)
         scaling_factor = (2.0 / sample_rate) / num_points_per_window
-        self.power = scaling_factor * squared_fft
-        self._fit_range = (self.frequency.min(), self.frequency.max())
-
-        # Store a variance for temporally blocked power spectra
-        self._variance = (
-            scaling_factor**2 * np.var(squared_fft_chunks, axis=0)
-            if len(squared_fft_chunks) > 1
-            else None
+        frequency = np.fft.rfftfreq(num_points_per_window, 1.0 / sample_rate)
+
+        return PowerSpectrum(
+            frequency=frequency,
+            power=scaling_factor * squared_fft,
+            sample_rate=sample_rate,
+            total_duration=data.size / sample_rate,
+            unit=unit,
+            num_points_per_block=len(squared_fft_chunks),
+            total_samples_used=num_points_per_window * (len(data) // num_points_per_window),
+            variance=(
+                scaling_factor**2 * np.var(squared_fft_chunks, axis=0)
+                if len(squared_fft_chunks) > 1
+                else None
+            ),
         )
 
-        # Store metadata
-        self.sample_rate = sample_rate
-        self.total_duration = data.size / sample_rate
-        self.num_points_per_block = len(squared_fft_chunks)
-        self.total_sampled_used = num_points_per_window * self.num_points_per_block
+    @property
+    def total_sampled_used(self) -> int:
+        return self.total_samples_used
 
     @property
-    def frequency_bin_width(self):
+    def frequency_bin_width(self) -> float:
         """Returns the frequency bin width of the spectrum"""
-        return self.sample_rate / self.total_sampled_used * self.num_points_per_block
+        return self.sample_rate / self.total_samples_used * self.num_points_per_block
+
+    def _apply_transforms(self, data, with_exclusions=True, downsample_data=True) -> np.ndarray:
```

Comment:
I think a little docstring would be helpful here as it's not apparent to those of us that haven't delved deep into the intricacies of this math what transform needs to be applied or what it means

Reply:
Fair point! I'll add a docstring.